alarm/nialarm.yml

```diff
@@ -0,0 +1,686 @@
+        description: Optional description text for a given alarm instance
+        type: string
+        example: Disk space on tester-1 is high. Disk usage was 95% when this alarm was created.
+      keywords:
+        description: Keywords associated with an alarm instance. Alarm instances can be tagged with keywords to make it easier to find them with queries.
+        type: array
+        items:
+          type: string
+        example:
+          - keyword1
+          - keyword2
+      notes:
+        description: A collection of notes for a given alarm instance. Notes are set by humans to record information about an alarm after it has been investigated.
+        type: array
+        items:
+          $ref: '#/definitions/AlarmNote'
+      properties:
+        description: Key-value-pair metadata associated with an alarm instance. Alarm instances can be tagged with properties so that they can be more easily found with queries. Additionally, alarm instance properties are used to expand message templates in the notification strategies associated with a given alarm instance.
+        type: object
+        additionalProperties:
+          type: string
+        example:
+          key1: value1
+      resourceType:
+        description: The type of resource associated with a given alarm instance
+        type: string
+        example: Tag
+  AlarmNote:
+    title: Alarm Note
+    description: Information about a particular alarm instance, such as a description of an investigation into an alarm's root cause.
+    required:
+      - note
+    properties:
+      note:
+        description: Information about a particular alarm instance, such as a description of an investigation into an alarm's root cause.
+        type: string
+        example: Temperature sensor was faulty
+      createdAt:
+        description: The time at which the note was created
+        type: string
+        example: '2019-01-14T23:15:53.7270539Z'
+      user:
+        description: The user who created the note
+        type: string
+        readOnly: true
+        example: Philip
+  AlarmTransitionType:
+    title: Alarm Transition Type
+    type: string
+    enum: [SET, CLEAR]
+  AlarmTransition:
+    title: Alarm Transition
+    description: A transition within a given alarm instance. Transitions can record changes to an alarm's *clear* field, as well as an alarm increasing or decreasing in severity.
+    type: object
+    required:
+      - transitionType
+    properties:
+      transitionType:
+        $ref: '#/definitions/AlarmTransitionType'
+      occurredAt:
+        description: The time at which the transition occurred.
+        type: string
+        example: '2019-01-14T23:15:53.7270539Z'
+      severityLevel:
+        description: The severity of the transition. Valid values for CLEAR transitions are [-1,-1]. Valid values for SET transitions are [1, 4].
+        type: integer
+        example: 2
+        default: 2
+        minimum: -1
+        maximum: 4
```

Comment:
We don't, but the we only have display strings like "Critical (4)" for these values in the UI, so I wanted to sort of encourage folks to only use those values without actually enforcing a hard limit. Let me know if you think the document should reflect what the API actually allows instead.

Reply:
A strict Swagger codegen could in theory throw an exception trying to use a value greater than 4 since that's what it says is the max. I don't know of any clients specifically that do that, but I believe the Python server codegen does. Since it's more of a "soft" maximum, maybe putting that in the description instead would be sufficient?